h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstPivot.java

```diff
@@ -0,0 +1,154 @@
+    // This is the sort then MRTask method.
+    // Create the target Frame
+    // Now sort on the index key, result is that unique keys will be localized
+    fr = fr.sort(new int[]{0});
+    final long[] classes = new VecUtils.CollectDomain().doAll(fr.vec(colIdx)).domain();
+    final int nClass = fr.vec(colIdx).isNumeric() ? classes.length : fr.vec(colIdx).domain().length;
+    String[] header = (String[]) ArrayUtils.addAll(new String[]{index}, fr.vec(colIdx).domain());
+    Frame initialPass = new pivotTask(fr,index,column,value).doAll(nClass+1, Vec.T_NUM, fr).outputFrame(null, header, null);
+    initialPass = initialPass.sort(new int[]{0});
+    // Collapse identical index rows even when index crosses over chunk boundaries
+    Frame secondPass = new pivotCleanup().doAll(nClass+1,Vec.T_NUM,initialPass).outputFrame(null,header,null);
+    Frame result = new Frame(secondPass.vec(0).makeCopy(null,fr.vec(indexIdx).get_type()));
+    result._key = Key.<Frame>make();
+    result.setNames(new String[]{index});
+    secondPass.remove(0);
+    result.add(secondPass);
+    initialPass.delete();
+    return new ValFrame(result);
+  }
+
+  protected class pivotCleanup extends MRTask<AstPivot.pivotCleanup>{
+    pivotCleanup() {}
+    @Override
+    public void map(Chunk[] cs, NewChunk[] nc) {
+      // skip past the first rows of the first index if we know that the previous chunk will run in here
+      long firstIdx =  cs[0].at8(0);
+      long globalIdx = cs[0].start();
+      int start = 0;
+      if (globalIdx > 0 && cs[0].vec().at8(globalIdx-1)==firstIdx){
+        while(firstIdx == cs[0].at8(start)) start++;
+      }
+      for (int i=start; i<cs[0]._len; i++) {
+        long currentIdx = cs[0].at8(i);
+        if (currentIdx != cs[0].at8(i+1)) {
+          nc[0].addNum(cs[0].at8(i));
+          for (int j=1;j<nc.length;j++) nc[j].addNum(cs[j].atd(i));
+          // were done here since we know the next row has a different index
+          continue;
+        }
+        // here we know we have to search ahead
+        int count = 1;
+        double[] newRow = new double[nc.length-1];
+        // start with a copy of the current row
+        for (int j = 0; j < nc.length-1; j++) newRow[j] = cs[j+1].atd(i);
+
+        while ( count + i < cs[0]._len && currentIdx == cs[0].at8(i + count) ) {
+          // merge the forward row, the newRow and the existing row
+          // here would be a good place to apply aggregating function
+          // for now we are aggregating by "first"
+          for (int j = 0; j < nc.length - 1; j++) {
+            if (Double.isNaN(newRow[j]) && !Double.isNaN(cs[j + 1].atd(i + count))) {
+              newRow[j] = cs[j + 1].atd(i + count);
+            }
+          }
+          // need to look if we need to go to next chunk
+          if (i + count == cs[0]._len - 1) {
+            Chunk nextChunk = cs[0].nextChunk();
+            if (nextChunk != null && currentIdx == nextChunk.at8(0)) {
```

Comment:
What will happen if one "index" spans more than 2 chunks? (eg. the index column is constant)?